pkg/db-repo/errors.go

```diff
@@ -54,3 +54,27 @@ func (e NoQueryResultsError) Unwrap() error {
 func IsNoQueryResultsError(err error) bool {
 	return errors.As(err, &NoQueryResultsError{})
 }
+
+type TooManyResultsError struct {
+	modelId string
+	err     error
+}
+
+func NewTooManyResultsError(found int, expected int, modelId string) TooManyResultsError {
+	return TooManyResultsError{
+		modelId: modelId,
+		err:     fmt.Errorf("found %d result(s), expected %d result(s)", found, expected),
```

Comment:
What do you need this additional error for? Isn't it sufficient to store `found` and `expected`? The Unwrap also doesn't make much sense, as you'll never be able to assert for this specific error as it's untyped